main.go

```diff
@@ -53,215 +60,346 @@ func NewClient(login, password string, httpClient *http.Client) *Client {
 	if httpClient == nil {
 		httpClient = &http.Client{
 			Transport: &http.Transport{
-				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+				MaxIdleConnsPerHost: 20,
+				TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
 			},
-			Jar: cookies,
+			Jar:     cookies,
+			Timeout: 1 * time.Minute,
 		}
 	}
 
 	cli := &Client{
-		login:    login,
-		password: password,
-		http:     httpClient,
+		username:   login,
+		password:   password,
+		httpClient: httpClient,
 	}
 
 	return cli
 }
 
-func (cli *Client) Login() error {
-	if cli.uid == "" {
-		uid, err := cli.getUid()
-		if err != nil {
-			return err
-		}
-
-		cli.uid = uid
-	}
-
-	payload := url.Values{
-		"goto":       {urlLoginSuccess},
-		"gotoOnFail": {urlLoginFail},
-		"org":        {"customer"},
-		"old-token":  {cli.login},
-		"IDToken2":   {cli.password},
-		"IDToken1":   {cli.uid},
+func (cli *Client) getDevices() DevicesInfo {
+	req, err := http.NewRequest(http.MethodGet, urlDevices, nil)
+	req.Header.Add("User-Agent", hUserAgent)
+	req.Header.Set("Accept", hContentTypeJson)
+	req.Header.Set("Authorization", basicAuthStr)
+	res, err := cli.httpClient.Do(req)
+	if err != nil {
+		log.Printf("req do err: %s", err)
```

Comment:
Oof, I don't think we can do anything if the method Do() has failed and should really interrupt the execution and return the error if the request was failed. It means you will also have to change the method signature to be `(DevicesInfo, error)`.

If the error is not nil the `res` will also be `nil` (since it's the way most of Go packages/funcs work) which means you will actually get a panic in the next lines where you work with `res` if the request was failed.